Пакулин Сергей

IIaKyJIuH opened the pull request.

IIaKyJIuH committed:
Позволит ли Хрюндель пройти все тесты сразу?

honest-hrundel commented:
:tomato: Пройдено тестов 20 из 26

IIaKyJIuH committed:
Nice try

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH committed:
what else should i do...

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH commented:
🔔

IIaKyJIuH commented:
Скажите, пожалуйста, что ещё за тесты не проходит?

IIaKyJIuH committed:
Finally???

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH committed:
Enough

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH committed:
just changed type of undefined

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH committed:
just checking if the problem in extra task

honest-hrundel commented:
:tomato: Пройдено тестов 21 из 23

IIaKyJIuH committed:
deleted my regular expression

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH committed:
just made it easy somehow

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH committed:
just added one sort func

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH committed:
just changed 'const' on 'let'

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH committed:
useless fixes

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH commented:
🔔

IIaKyJIuH commented:
don`t know what to do...

IIaKyJIuH committed:
Made it easy v2

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH commented:
🔔

IIaKyJIuH committed:
search -> indexOf, slice -> substring

honest-hrundel commented:
:green_apple: Пройдено тестов 26 из 26

honest-hrundel commented:
@FrodoTheTrue обрати внимание: решено доп. задание

FrodoTheTrue commented on phone-book.js:
```diff
@@ -61,8 +133,20 @@ function importFromCsv(csv) {
     // Добавляем в телефонную книгу
     // Либо обновляем, если запись с таким телефоном уже существует
+    const contacts = csv.split('\n');
+    let kills = 0;
```
kills?

FrodoTheTrue commented on phone-book.js:
```diff
@@ -61,8 +133,20 @@ function importFromCsv(csv) {
+    let kills = 0;
+    for (let contact of contacts) {
+        contact = contact.split(';');
+        let data = {
```
`const [name, phone, email] = contact.split(';');`

FrodoTheTrue commented on phone-book.js:
```diff
@@ -39,7 +70,41 @@ function update(phone, name, email) {
+    return dataToDelete.length;
+}
+
+function everythingFrom(array) {
```
совсем непонятное название функции

FrodoTheTrue commented on phone-book.js:
```diff
@@ -9,7 +9,24 @@ const isStar = true;
+    return isString(value) && value !== '';
+}
+
+function checkInput(phone, name) {
```
`isCorrectInput` тогда

FrodoTheTrue commented on phone-book.js:
```diff
@@ -9,7 +9,24 @@ const isStar = true;
  * Телефонная книга
  */
-let phoneBook;
+let phoneBook = [];
```
здесь предпочтительнее делать через словарь

FrodoTheTrue commented on phone-book.js:
```diff
@@ -19,7 +36,14 @@ let phoneBook;
  */
 function add(phone, name, email) {
+    if (checkInput(phone, name) &&
+        typeof(phoneBook[phone]) === 'undefined') {
```
на undefined проверяем `if (!phoneBook[phone]) { ... }`

FrodoTheTrue replied:
а лучше прямо фунции указать по умолчанию пустой email:
`function add(phone, name, email='') {`

FrodoTheTrue commented on phone-book.js:
```diff
@@ -39,7 +70,41 @@ function update(phone, name, email) {
+    let dataToDelete = find(query);
+    for (let data of dataToDelete) {
+        // приведём номер к простому виду и избавимся в начале строки от 7
+        let phone = data.split(', ')[1].replace(/[- +()]/g, '').substring(1);
```
везде где можно стоит использовать `const`

FrodoTheTrue commented on phone-book.js:
```diff
@@ -39,7 +70,41 @@ function update(phone, name, email) {
+function getAllBy(element) {
+    let result = [];
+    for (let key of Object.keys(phoneBook)) {
+        if (phoneBook[key][0].indexOf(element) !== -1 || key.indexOf(element) !== -1 ||
```
через `include` будет чище

IIaKyJIuH committed:
Всё изменил

kills = passed;
сделал сплит сразу в массив;
everythingFrom() = getAllFrom();
checkInput()=isCorrectInput();
Заменил массив phoneBook на словарь;
Упростил проверку на undefined;
Везде, где можно, теперь использую const;
Проверку запроса делаю через include;

honest-hrundel commented:
:tomato: Пройдено тестов 22 из 26

IIaKyJIuH committed:
Некоторые правки

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH committed:
Ещё доработки

honest-hrundel commented:
:tomato: Пройдено тестов 24 из 26

IIaKyJIuH commented:
🔔

IIaKyJIuH commented:
Скажите, пожалуйста, в чём проблема кода?

IIaKyJIuH committed:
Добавил проверку в findAndRemove()

honest-hrundel commented:
:tomato: Пройдено тестов 25 из 26

IIaKyJIuH committed:
Возвращаю [], а не {} в find()

honest-hrundel commented:
:green_apple: Пройдено тестов 26 из 26

honest-hrundel commented:
@FrodoTheTrue обрати внимание: решено доп. задание

IIaKyJIuH commented:
@FrodoTheTrue , точно ли у меня должно быть "closed-after-deadline" и "half-accepted", я же сдал до 25 и закрыл до 1-го?
